MoltenVK/MoltenVK/Commands/MVKCmdAccelerationStructure.mm

```diff
@@ -0,0 +1,200 @@
+                                                             VkAccelerationStructureKHR         srcAccelerationStructure,
+                                                             uint64_t                           dstAddress,
+                                                             VkCopyAccelerationStructureModeKHR copyMode) {
+    _dstAddress = dstAddress;
+    _copyMode = copyMode;
+    
+    MVKAccelerationStructure* mvkSrcAccStruct = (MVKAccelerationStructure*)srcAccelerationStructure;
+    _srcAccelerationStructure = mvkSrcAccStruct->getMTLAccelerationStructure();
+    
+    _dstBuffer = _mvkDevice->getBufferAtAddress(_dstAddress);
+    return VK_SUCCESS;
+}
+                                        
+void MVKCmdCopyAccelerationStructureToMemory::encode(MVKCommandEncoder* cmdEncoder) {
+    id<MTLAccelerationStructureCommandEncoder> accStructEncoder = cmdEncoder->getMTLAccelerationStructureEncoder(kMVKCommandUseNone);
+    _mvkDevice = cmdEncoder->getDevice();
+    
+    if(_copyMode != VK_COPY_ACCELERATION_STRUCTURE_MODE_SERIALIZE_KHR || !_dstBuffer->getDeviceMemory()->isMemoryHostAccessible()){
+        return;
+    }
+    
+    memcpy(_dstBuffer->getDeviceMemory()->getHostMemoryAddress(), (void*)_srcAccelerationStructure, sizeof(_srcAccelerationStructure));
+}
+
+#pragma mark -
+#pragma mark MVKCmdCopyMemoryToAccelerationStructure
+
+VkResult MVKCmdCopyMemoryToAccelerationStructure::setContent(MVKCommandBuffer* cmdBuff,
+                                                                uint64_t srcAddress,
+                                                                VkAccelerationStructureKHR dstAccelerationStructure,
+                                                                VkCopyAccelerationStructureModeKHR copyMode) {
+    _srcAddress = srcAddress;
+    _copyMode = copyMode;
+    
+    _srcBuffer = _mvkDevice->getBufferAtAddress(_srcAddress);
+    
+    MVKAccelerationStructure* mvkDstAccStruct = (MVKAccelerationStructure*)dstAccelerationStructure;
+    _dstAccelerationStructure = mvkDstAccStruct->getMTLAccelerationStructure();
+    return VK_SUCCESS;
+}
+
+void MVKCmdCopyMemoryToAccelerationStructure::encode(MVKCommandEncoder* cmdEncoder) {
+    id<MTLAccelerationStructureCommandEncoder> accStructEncoder = cmdEncoder->getMTLAccelerationStructureEncoder(kMVKCommandUseNone);
+    _mvkDevice = cmdEncoder->getDevice();
+    
+    if(_copyMode != VK_COPY_ACCELERATION_STRUCTURE_MODE_DESERIALIZE_KHR){
+        return;
+    }
+    
+    void* serializedAccStruct = _srcBuffer->getHostMemoryAddress();
+    if(!serializedAccStruct){
+        return; // Should I remove this? For this to work, the memory can't be device only, but the spec does not seem to restrict this
+    }
+
+    MVKAccelerationStructure* mvkSrcAccStruct = (MVKAccelerationStructure*)serializedAccStruct;
+    id<MTLAccelerationStructure> srcAccelerationStructure = mvkSrcAccStruct->getMTLAccelerationStructure();
+    
+    [accStructEncoder
```

Comment:
Er, uh, you're still assuming the destination resource has a `VkAccelerationStructure` created at that address. You'll need to make an analogous change here to the one you need to make for `vkCmdCopyAccelerationStructureToMemory()` above.